nltk/parse/dependencygraph.py

```diff
@@ -543,6 +543,109 @@ def nx_graph(self):
+        ...                               u'CharacterOffsetEnd': u'10',
+        ...                               u'Lemma': u'love',
+        ...                               u'PartOfSpeech': u'VBZ'}],
+        ...                             [u'Mary',
+        ...                              {u'CharacterOffsetBegin': u'11',
+        ...                               u'CharacterOffsetEnd': u'15',
+        ...                               u'Lemma': u'Mary',
+        ...                               u'PartOfSpeech': u'NNP'}],
+        ...                             [u'.',
+        ...                              {u'CharacterOffsetBegin': u'15',
+        ...                               u'CharacterOffsetEnd': u'16',
+        ...                               u'Lemma': u'.',
+        ...                               u'PartOfSpeech': u'.'}]]}
+        >>> list(stanfordjson2conll(stanfordjson)) # stanford-corenlp-full-2015-04-20.
+        [(2, '_', 'loves', 'love', 'VBZ', 'VBZ', '_', '0', 'ROOT', '_', '_'), (1, '_', 'John', 'John', 'NNP', 'NNP', '_', '2', 'nsubj', '_', '_'), (3, '_', 'Mary', 'Mary', 'NNP', 'NNP', '_', '2', 'dobj', '_', '_'), (4, '_', '.', '.', '.', '.', '_', '2', 'punct', '_', '_')]
+        
+        :param sentence: The JSON output from Stanford CoreNLP tool.
+        :type sentence: dict
+        :rtype: generator
+        """
+        for node in jsonparse['dependencies']:
+            # Parses the node for different stanford-corenlp versions.
+            if len(node) == 5: # stanford-corenlp-full-2015-04-20.
+                rel, _, head, word, n = node
+            elif len(node) == 3: # For stanford-corenlp-full-2014-08-27.
+                rel, head, word = node
+                word, n = word.split('-')
+                _ , head = head.split('-') 
+            # Access the words, POS and lemma. 
+            n = int(n)
+            word_info = jsonparse['words'][n - 1][1]
+            tag = word_info['PartOfSpeech']
+            lemma = word_info['Lemma']
+            if rel == 'root':
+                # NLTK expects that the root relation is labelled as ROOT!
+                rel = 'ROOT'
+            # Hack: Return values we don't know as '_'.
+            #       Also, consider tag and ctag to be equal.
+            # n is used to sort words as they appear in the sentence.
+            yield n, '_', word, lemma, tag, tag, '_', head, rel, '_', '_'
+
+    @staticmethod
+    def fromstanfordjson(jsonparse):
+        """
+        This function takes a JSON output from Stanford CoreNLP and converts 
+        it into a DependencyGraph object.
+    
+        :param jsonparse: The JSON output from Stanford CoreNLP tool.
+        :type sentence: dict
+        :rtype: DependencyGraph
+        """
+        # Converts Stanford JSON outputs to CONLL
+        conllparse = DependencyGraph.stanfordjson2conll(jsonparse)
+        return DependencyGraph([' '.join(items[1:]) for 
```

Comment:
set ``cell_separator=' '`` CoreNLP uses unbreakable space inside of a token, which might break DependencyGraph.